packages/aws-rfdk/docs/upgrade/upgrading-0.38.md

```diff
@@ -0,0 +1,254 @@
+# Upgrading to RFDK v0.38.x or Newer
+
+Starting in RFDK v0.38.0, [Deadline Secrets Management](https://docs.thinkboxsoftware.com/products/deadline/10.1/1_User%20Manual/manual/secrets-management/deadline-secrets-management.html) will be enabled on the
+`Repository` and `RenderQueue` constructs by default when using Deadline 10.1.19 and later. This can cause some issues with existing farm configurations because Deadline Secrets Management requires:
+
+- The internal protocol on the Render Queue to be HTTPS
+- The external protocol on the Render Queue to be HTTPS
+
+Additionally, if you plan on upgrading to RFDK 0.38 or newer and Deadline 10.1.19 or newer, but do **not** want to enable Deadline Secrets Management, you may have to make changes to your RFDK application so that
+RFDK does not automatically configure Deadline Secrets Management on your farm.
+
+## Upgrading RFDK and Enabling Deadline Secrets Management
+
+---
+
+_This section describes how to upgrade your RFDK farm while also enabling Deadline Secrets Management. If you would **not** like to enable Deadline Secrets Management, please skip to the
+next section "Upgrading RFDK Without Enabling Deadline Secrets Management"._
+
+---
+
+Your `RenderQueue` will use HTTPS for both internal and external protocols for RFDK 0.38.0 if **EITHER** of the following are true:
+
+1. You **HAVE NOT** specified the `trafficEncryption` (TypeScript) / `traffic_encryption` (Python) property
+1. You **HAVE** specified the `trafficEncryption` (TypeScript) / `traffic_encryption` (Python) property **AND BOTH of the following are true**:
+    1. **EITHER** of the following are true:
+        1. You **HAVE NOT** specified the `trafficEncryption.internalProtocol` (TypeScript) / `traffic_encryption.internal_protocol` (Python)
+        1. You **HAVE** specified the `trafficEncryption.internalProtocol` (TypeScript) / `traffic_encryption.internal_protocol` (Python) and it is set to `ApplicationProtocol.HTTPS`
+    1. **EITHER** of the following are true:
+        1. You **HAVE NOT** specified the `trafficEncryption.externalTLS` (TypeScript) / `traffic_encryption.external_tls` (Python)
+        1. You **HAVE NOT** specified `trafficEncryption.externalTLS.enabled` (TypeScript) / `traffic_encryption.external_tls.enabled` (Python)
+        1. You **HAVE** specified `trafficEncryption.externalTLS.enabled` (TypeScript) / `traffic_encryption.external_tls.enabled` (Python) and it is set to `true` (TypeScript) / `True` (Python)
```

Comment:
This seems really wordy way of explaining that HTTPS is on by default or uses the configured value. It also doesn't explain why we care about HTTPS; it does get explained above, but I feel like we should re-iterate it since this is a different section. Can't we sum it up like this:

```
The first step to making sure we're ready to enable Secrets Management is to make sure we have HTTPS 
enabled on the `RenderQueue` for both the internal and external protocols. If you are upgrading from a version 
prior to 0.37, you should refer to the [v0.37 upgrade doc](./upgrading-0.37.md) for details about how the 
`RenderQueue` will default to using HTTPS now.

If any of the optional fields for configuring the protocol the `RenderQueue` uses aren't set, it will default to using HTTPS. 
* Unless `trafficEncryption.internalProtocol` is set to `ApplicationProtocol.HTTP`, the internal protocol will be HTTPS.
* Unless `trafficEncryption.externalTLS.enabled` is set to `false`, the external protocol will be HTTPS.
```

Reply:
Let's leave what you currently have as-is, but I still think adding the first little blurb I wrote out would be helpful:

```
The first step to making sure we're ready to enable Secrets Management is to make sure we have HTTPS 
enabled on the `RenderQueue` for both the internal and external protocols. If you are upgrading from a version 
prior to 0.37, you should refer to the [v0.37 upgrade doc](./upgrading-0.37.md) for details about how the 
`RenderQueue` will default to using HTTPS now.
```

Reply:
Sounds good, added